spec/requests/api/v1/targets/destroy_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+describe 'Destroy Target', type: :request do
+  subject { delete api_v1_target_path(target), headers:, as: :json }
+
+  let(:user) { create(:user) }
+  let(:target) { create(:target, user:) }
+  let(:headers) { user.create_new_auth_token }
+
+  context 'when success' do
+    it 'should return a sucessfull response' do
+      subject
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it 'should delete the target' do
+      subject
+      expect(Target.count).to eq(0)
```

Comment:
I think is better to check  that only change the user targets, not all of them

```suggestion
      expect {
        subject
      }.to change(user.targets, :count).by(-1)
```